packages/commonwealth/server/api/integration-router.ts

```diff
@@ -0,0 +1,91 @@
+import { express } from '@hicommonwealth/adapters';
+import { ChainEvents, Thread } from '@hicommonwealth/model';
+import { RequestHandler, Router, raw } from 'express';
+
+// TODO: remove as we migrate to tRPC commands
+import DatabaseValidationService from 'server/middleware/databaseValidationService';
+import { deleteBotCommentHandler } from 'server/routes/comments/delete_comment_bot_handler';
+import { updateCommentHandler } from 'server/routes/comments/update_comment_handler';
+import { createThreadCommentHandler } from 'server/routes/threads/create_thread_comment_handler';
+import { deleteBotThreadHandler } from 'server/routes/threads/delete_thread_bot_handler';
+import { updateThreadHandler } from 'server/routes/threads/update_thread_handler';
+import { ServerControllers } from 'server/routing/router';
+
+const PATH = '/api/integration';
+
+function build(
+  controllers: ServerControllers,
+  validator: DatabaseValidationService,
+) {
+  // Async middleware wrappers
+  const isBotUser: RequestHandler = (req, res, next) => {
+    validator.validateBotUser(req, res, next).catch(next);
+  };
+  const isAuthor: RequestHandler = (req, res, next) => {
+    validator.validateAuthor(req, res, next).catch(next);
+  };
+
+  const router = Router();
+  router.use(express.statsMiddleware);
+
+  // Chain Events integration
+  router.post(
+    '/chainevent/ChainEventCreated/:id',
+    raw({ type: '*/*', limit: '10mb', inflate: true }),
+    (req, _, next) => {
+      ChainEvents.verifyAlchemySignature(req);
+      return next();
+    },
+    // parse body as JSON (native express.json middleware doesn't work here)
+    (req, _, next) => {
```

Comment:
@timolegros do we need this body parser? the other route seems to parse json bodies ok

Reply:
Yes this is necessary. The other routes don't have `raw({ type: '*/*', limit: '10mb', inflate: true })` as the first handler.